src/proxy-ui-api/frontend/src/views/AddClient/ClientDetailsPage.vue

```diff
@@ -0,0 +1,226 @@
+            v-model="memberCode"
+            data-test="member-code-input"
+          ></v-text-field>
+        </ValidationProvider>
+      </div>
+
+      <div class="row-wrap">
+        <FormLabel labelText="wizard.subsystemCode" helpText="wizard.client.subsystemCodeTooltip" />
+
+        <ValidationProvider name="addClient.subsystemCode" rules="required" v-slot="{ errors }">
+          <v-text-field
+            class="form-input"
+            type="text"
+            :error-messages="errors"
+            v-model="subsystemCode"
+            data-test="subsystem-code-input"
+          ></v-text-field>
+        </ValidationProvider>
+      </div>
+      <div v-if="duplicateClient" class="duplicate-warning">{{$t('wizard.client.memberExists')}}</div>
+      <div class="button-footer">
+        <div class="button-group">
+          <large-button outlined @click="cancel" data-test="cancel-button">{{$t('action.cancel')}}</large-button>
+        </div>
+        <large-button
+          @click="done"
+          :disabled="invalid || duplicateClient"
+          data-test="next-button"
+        >{{$t('action.next')}}</large-button>
+      </div>
+    </ValidationObserver>
+
+    <SelectClientDialog
+      :dialog="showSelectClient"
+      :selectableClients="selectableClients"
+      @cancel="showSelectClient = false"
+      @save="saveSelectedClient"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { mapGetters } from 'vuex';
+import FormLabel from '@/components/ui/FormLabel.vue';
+import LargeButton from '@/components/ui/LargeButton.vue';
+import SelectClientDialog from './SelectClientDialog.vue';
+import { Client } from '@/types';
+
+import { ValidationProvider, ValidationObserver } from 'vee-validate';
+
+export default Vue.extend({
+  components: {
+    FormLabel,
+    LargeButton,
+    ValidationObserver,
+    ValidationProvider,
+    SelectClientDialog,
+  },
+  computed: {
+    ...mapGetters(['reservedClients', 'selectableClients']),
+
+    memberClass: {
+      get(): string {
+        return this.$store.getters.memberClass;
+      },
+      set(value: string) {
+        this.$store.commit('setMemberClass', value);
+      },
+    },
+
+    memberCode: {
+      get(): string {
+        return this.$store.getters.memberCode;
+      },
+      set(value: string) {
+        this.$store.commit('setMemberCode', value);
+      },
+    },
+
+    subsystemCode: {
+      get(): string {
+        return this.$store.getters.subsystemCode;
+      },
+      set(value: string) {
+        this.$store.commit('setSubsystemCode', value);
+      },
+    },
+
+    selectedMember: {
+      get(): Client {
+        return this.$store.getters.selectedMember;
+      },
+      set(value: Client) {
+        this.$store.commit('setMember', value);
+      },
+    },
+
+    duplicateClient(): boolean {
+      if (!this.memberClass || !this.memberCode || !this.subsystemCode) {
+        return false;
+      }
+
+      if (
```

Comment:
I think you should extract the contents of this if condition and make a helper function out of it instead. Optionally just extract the lambda function

Reply:
Moved this to helpers